homeassistant/components/decora_wifi/__init__.py

```diff
@@ -1 +1,129 @@
-"""The decora_wifi component."""
+"""Support for the myLeviton decora_wifi component."""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from .common import (
+    DecoraWifiCommFailed,
+    DecoraWifiLoginFailed,
+    DecoraWifiPlatform,
+    decorawifisessions,
+)
+from .const import CONF_OPTIONS, DEFAULT_SCAN_INTERVAL, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+DECORAWIFI_HOST_SCHEMA = vol.Schema({vol.Required(CONF_HOST): cv.string})
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): cv.string,
+                    vol.Required(CONF_PASSWORD): cv.string,
+                    vol.Optional(CONF_SCAN_INTERVAL, default=120): cv.positive_int,
+                },
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Decora Wifi component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if DOMAIN not in config:
+        return True
+
+    # Trigger import config flow
+    conf = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={
+                    CONF_USERNAME: conf[CONF_USERNAME],
+                    CONF_PASSWORD: conf[CONF_PASSWORD],
+                },
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry):
+    """Set up Decora Wifi from a config entry."""
+
+    conf_data = dict(entry.data)
+    email = conf_data[CONF_USERNAME]
+    password = conf_data[CONF_PASSWORD]
+
+    conf_data[CONF_OPTIONS] = {}
+    # If no scan interval option was set in the config entry, use the default.
+    conf_data[CONF_OPTIONS].update(
+        {
+            CONF_SCAN_INTERVAL: dict(entry.options).get(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+            )
+        }
+    )
```

Comment:
```suggestion
    # If no scan interval option was set in the config entry, use the default.
    conf_data[CONF_OPTIONS] = {CONF_SCAN_INTERVAL: dict(entry.options).get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)}
```

No need to update it if you are going to clear it right before.  

Will need `black`

Reply:
Done in next commit.